apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette.component.ts

```diff
@@ -22,28 +26,60 @@ import { DotPaletteContentletsComponent } from './dot-palette-contentlets/dot-pa
                     transform: 'translateX(0%)'
                 })
             ),
-            transition('* => *', animate('200ms ease-in')),
+            transition('* => *', animate('200ms ease-in'))
         ])
     ]
 })
-export class DotPaletteComponent {
-    @Input() items: DotCMSContentType[] = [];
+export class DotPaletteComponent implements OnInit {
+    @Input() set items(items: DotCMSContentType[]) {
+        if (items != null && items.length) {
+            this.store.loadContentTypes(items);
+        }
+    }
     @Input() languageId: string;
-    contentTypeVariable = '';
-    stateContentlet = 'contentlet:out';
+    vm$: Observable<any> = this.store.vm$;
 
     @ViewChild('contentlets') contentlets: DotPaletteContentletsComponent;
     @ViewChild('contentTypes') contentTypes: DotPaletteContentTypeComponent;
 
+    constructor(private store: DotPaletteStore) {}
+
+    ngOnInit(): void {
+        this.store.setLanguageId(this.languageId);
+    }
+
     /**
-     * Sets value on contentTypeVariable variable to show/hide components on the UI
+     * Sets value on store variables to show/hide components on the UI
      *
      * @param string [variableName]
      * @memberof DotPaletteContentletsComponent
      */
     switchView(variableName?: string): void {
-        this.contentTypeVariable = variableName ? variableName : '';
-        this.stateContentlet = variableName ? 'contentlet:in' : 'contentlet:out';
+        const viewContentlet = variableName ? 'contentlet:in' : 'contentlet:out';
+
+        this.store.setViewContentlet(viewContentlet);
+        this.store.setFilter('');
+        this.store.loadContentlets(variableName);
```

Comment:
```suggestion
        this.store.switchView(...);
```

And do all the other things internally? We only call all those here.